### PR TITLE
Add `TensorBoardWriter`

### LINKED ISSUE #28: Do you have a plan to support TensorBoard
PyTorch supports TensorBoard naitively in `torch.utils.tensorboard`. But ppe's `LogReport` doesn't because tensorbord writers don't satisfy the requirement.

https://github.com/pfnet/pytorch-pfn-extras/blob/b2ddc4570f655bfe00f1d9f7dbb64fd4654a73c4/pytorch_pfn_extras/training/extensions/log_report.py#L57-L60

So my question is would ppe support it? Or do we have any other way to do it?

### PR DESCRIPTION
**Pull request by emcastillo**
Closes #28 

Creating the `LogReport` extension with the kwarg `writer=TensorBoardWriter()`
should automatically report the metrics to tensorboard.

Users can also use this writer to report their own metrics in their modules.

### CODE REVIEW

**Comment by emcastillo** [comment]
Thanks a lot to @belltailjp for his help when doing the writing system, this allowed us to solve this pretty easily.

File: pytorch_pfn_extras/writing.py [hunk 1]
```diff
@@ -500,3 +500,21 @@ def create_queue(self):
...
     def create_consumer(self, q):
         return multiprocessing.Process(target=self.consume, args=(q,))
+
+
+class TensorBoardWriter(object):
+    def __init__(self, savefun=None, fs=None, out_dir=None, **kwds):
+        import torch.utils.tensorboard
+        self._writer = torch.utils.tensorboard.SummaryWriter()
+
+    def __call__(self, filename, out_dir, target, *, savefun=None):
```
Comment (emcastillo) [inline pytorch_pfn_extras/writing.py:510]: __call__ and __init__ attributes are just to keep compatibility.

**Comment by emcastillo** [comment]
I tested it with tensorboard and I confirm it works.
Changed the `ppe.training.extensions.LogReport()` in the mnist example to
`ppe.training.extensions.LogReport(writer=ppe.writing.TensorBoardWriter())`

File: pytorch_pfn_extras/writing.py [hunk 2]
```diff
@@ -500,3 +500,21 @@ def create_queue(self):
 
     def create_consumer(self, q):
         return multiprocessing.Process(target=self.consume, args=(q,))
+
+
+class TensorBoardWriter(object):
+    def __init__(self, savefun=None, fs=None, out_dir=None, **kwds):
+        import torch.utils.tensorboard
+        self._writer = torch.utils.tensorboard.SummaryWriter()
```
Comment (dhgrs) [inline pytorch_pfn_extras/writing.py:508]: `torch.utils.tensorboard.SummaryWriter()` is initialized with the default arguments. So the log file is saved in `runs/CURRENT_DATETIME_HOSTNAME` as described in the document.
https://pytorch.org/docs/stable/tensorboard.html#torch.utils.tensorboard.writer.SummaryWriter

How about use `out_dir` to initialize it?

**Comment by emcastillo** [comment]
Just pushed a change with support for `out_dir` and directly specifying the keyword args to the underlying `SummaryWriter`, also a test was added

**Comment by kmaehashi** [comment]
I confirmed that `tensorboard --logdir runs` correctly displayed the graph.

**Pull request merged**